### PR TITLE
Keyless verification command line flags.

### LINKED ISSUE #175: Feature Request: Add `kwctl` flags for keyless verification
### Is your feature request related to a problem?

_No response_

### Solution you'd like

For subcommands that perform verification (`kwctl {verify, pull, run}`), add flags for keyless verification. These flags would be like present `--verification-key` etc, but for:
- cert email checks (like cosign does)
- github owner

These flags will get processed by `build_verification_settings_from_flags()` to create a valid `VerificationConfig` that can be used.

### Alternatives you've considered

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by jvanz**
# Description

Adds the command line flags necessary to allow the users to verify the
keyless signatures. The flags allow the user to check signatures created
within Github actions and any other keyless signature.

Fix #175 

This PR depends on #169. Once it's merged, I'll rebase this one on top of main.

### CODE REVIEW

File: src/cli.rs [hunk 1]
```diff
@@ -93,6 +93,38 @@ pub fn build_cli() -> Command<'static> {
...
+                    .takes_value(true)
+                    .help("Github owner expected in the certificates generated in CD pipelines")
+                )
+                .arg(
+                    Arg::new("github-repo")
+                    .long("github-repo")
+                    .multiple_occurrences(false)
+                    .number_of_values(1)
+                    .takes_value(true)
+                    .help("Github repository expected in the certificates generated in CD pipelines")
```
Comment (flavio) [inline src/cli.rs:126]:
```suggestion
                    .help("GitHub repository expected in the certificates generated in CD pipelines")
```

File: src/cli.rs [hunk 2]
```diff
@@ -158,6 +190,38 @@ pub fn build_cli() -> Command<'static> {
...
+                    .takes_value(true)
+                    .help("Expected OIDC issuer in Fulcio certificates")
+                )
+                .arg(
+                    Arg::new("github-owner")
+                    .long("github-owner")
+                    .multiple_occurrences(false)
+                    .number_of_values(1)
+                    .takes_value(true)
+                    .help("Github owner expected in the certificates generated in CD pipelines")
```
Comment (flavio) [inline src/cli.rs:215]:
```suggestion
                    .help("GitHub owner expected in the certificates generated in CD pipelines")
```

File: src/cli.rs [hunk 3]
```diff
@@ -158,6 +190,38 @@ pub fn build_cli() -> Command<'static> {
...
+                )
+                .arg(
+                    Arg::new("github-repo")
+                    .long("github-repo")
+                    .multiple_occurrences(false)
+                    .number_of_values(1)
+                    .takes_value(true)
+                    .help("Github repository expected in the certificates generated in CD pipelines")
```
Comment (flavio) [inline src/cli.rs:223]:
```suggestion
                    .help("GitHub repository expected in the certificates generated in CD pipelines")
```

File: src/cli.rs [hunk 4]
```diff
@@ -290,6 +354,38 @@ pub fn build_cli() -> Command<'static> {
...
+                    .takes_value(true)
+                    .help("Expected OIDC issuer in Fulcio certificates")
+                )
+                .arg(
+                    Arg::new("github-owner")
+                    .long("github-owner")
+                    .multiple_occurrences(false)
+                    .number_of_values(1)
+                    .takes_value(true)
+                    .help("Github owner expected in the certificates generated in CD pipelines")
```
Comment (flavio) [inline src/cli.rs:379]:
```suggestion
                    .help("GitHub owner expected in the certificates generated in CD pipelines")
```

File: src/main.rs [hunk 5]
```diff
@@ -451,17 +451,74 @@ fn build_verification_options_from_flags(
...
+    {
         return Err(anyhow!(
-            "Intending to verify annotations, but no verification keys were passed"
+            "Intending to verify annotations, but no verification keys, OIDC issuer or Github owner were passed"
         ));
     }
+
+    if github_repo.is_some() && github_owner.is_none() {
+        return Err(anyhow!(
+            "Intending to verify Github actions signature, but the repository owner is missing."
```
Comment (flavio) [inline src/main.rs:492]:
```suggestion
            "Intending to verify GitHub actions signature, but the repository owner is missing."
```

**Review by flavio — COMMENTED** [review]
LGTM, I left some minor suggestions

**Comment by flavio** [comment]
@jvanz you allowed maintainers to write to this branch, I took advantage of that to perform the following actions:

* Rebase against `main` - the PR you built on top of has been merged
* Fix e2e test failures

I'll let the unit tests run, then I'll merge it as soon as everything is green

**Comment by jvanz** [comment]
> @jvanz you allowed maintainers to write to this branch, I took advantage of that to perform the following actions:
> 
>     * Rebase against `main` - the PR you built on top of has been merged
> 
>     * Fix e2e test failures
> 
> 
> I'll let the unit tests run, then I'll merge it as soon as everything is green

Thanks!

**Pull request merged**